### PR TITLE
[Google Drive] Handle failures when fetching file ACLs

### PR DESCRIPTION
**Pull request by jedrazb**
## Closes https://github.com/elastic/connectors-python/issues/1473

In certain scenarios the `permissions.list` endpoint will result in `403 insufficientFilePermissions` error. 

For example: someone else (e.g. from outside org) shares a file with you, you then go to drive UI to the "shared with me" tab and share this with the SA with view-only access. This chain of permissions is insufficient according to google to fetch ACL for that file. (Surprisingly the editor access allows to fetch that). See full scenario described in: https://github.com/elastic/connectors-python/issues/1474

Before, the whole sync would fail if we failed to fetch permissions even for a single file. In this PR I'm adding a logic to handle this gracefully and log problematic filenames and corresponding exceptions. 

Also, for the sake of completeness I added `https://www.googleapis.com/auth/drive.metadata.readonly` scope to drive client - to achieve parity with scopes used by older Workplace Search connector and this connector. 

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: connectors/sources/google_drive.py [hunk 1]
```diff
@@ -1100,9 +1103,16 @@ async def prepare_file(self, file, paths):
...
             permissions = file.get("permissions", [])
             if not permissions:
-                permissions = await self._get_permissions_on_shared_drive(
-                    file_id=file_id
-                )
+                try:
+                    permissions = await self._get_permissions_on_shared_drive(
+                        file_id=file_id
+                    )
+                except HTTPError as exception:
```
Comment (vidok) [inline connectors/sources/google_drive.py:1110]: not blocking: I wonder if some of the HTTPError exceptions can be fixed by retrying the request or reauthenticate it?

Comment (jedrazb) [reply connectors/sources/google_drive.py:1110]: > I wonder if some of the HTTPError exceptions can be fixed by retrying the request
Nope, we already have a re-try logic at a lower level. This `try/except` method is only supposed to gracefully handle failure scenarios after all retries are exhausted. The underlying issue is with scenario where user doesn't grant sufficient permissions to a file, enable dls, and so permissions fetching will always fail. 

> if you handle only a specific case like 403 error then it's worth adding an extra `try:` where you log that specific error.
Yes, exactly I only handle 403s in this clause. I can't really add extra logging at the route of the exception as this is handled by a library. Not sure if I understand what you mean here, can you elaborate. For context the logged `exception` will print the status code, and a detailed error message.

Comment (vidok) [reply connectors/sources/google_drive.py:1110]: > I can't really add extra logging at the root of the exception as this is handled by the aiogoogle library. Not sure if I understand what you mean here, can you elaborate.

what I mean is: 
```python
   try:
    .....
   except HTTPError as e:
     if e.reg.status == 403:
       self._logger.warning("Unable to fetch permissions due to 403 error. Exception {exception}")
    else:
      self._logger.error("Unable to fetch permissions due exception {exception}")
```

In this case, we can use different log levels or even handle those errors differently if we need. 
"Need" - is the main question here :)

**Review by vidok — APPROVED** [review]

File: connectors/sources/google_drive.py (hunk 1 above)
Comment (jedrazb) [reply connectors/sources/google_drive.py:1110]: That's a good point, we might want to make the non-403s more visible to the user. I updated the PR! :)

**Review by vidok — APPROVED** [review]

**Pull request merged**